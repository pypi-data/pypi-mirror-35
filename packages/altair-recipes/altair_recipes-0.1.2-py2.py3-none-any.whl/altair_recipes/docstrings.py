"""Reusable docstrings elements."""
from inspect import getfullargspec

docstrings = dict(
    data="""data: Altair Data or pandas DataFrame or csv or json file URL
    The data from which the statistical graphics is being generated""",
    column="""column: str or other column selector
    The column containing the data to be used in the graphics""",
    x="""x: str or other columns selector
    The column containing the data associated with the horizontal dimension""",
    y="""y: str or other column selector
    The column containing the data associated with the vertical dimension""",
    columns="""columns: str or list thereof or other column selector
    The column or columns to be used in the graphics""",
    group_by="""group_by: str or other column selector
    The column to be used to group the data when in long form. When group_by is
    specified columns should point to a single column""",
    return_object="""altair.Chart_area
    The graphics object""",
    mark="""mark: dictionary
    Additional arguments to pass to the mark method of Chart""",
    encoding="""encoding: dictionary
    Additional arguments to the encode method of Chart""",
    properties="""properties: dictionary
    Additional arguments to the properties method of Chart""",
    returns="""altair.Chart
    The chart described in the summary""")


def make_docstring(func, summary, additional_params={}):
    """Make docstrings from simple reusable parts.

    Parameters
    ----------
    summary : str
        The docstring summary.
    additional_params : dict
        A list of parameter docstrings. If present in the .docstrings.Docstring
        enum, will be replaced by corresponding value (default meaning for that
        argument)


    Returns
    -------
    str
        A docstring. Assign to __doc__ of desired object.

    """
    docstrings_ = docstrings.copy()
    docstrings_.update(additional_params)
    params = list(map(lambda x: docstrings_[x], getfullargspec(func).args))
    returns = docstrings['returns']
    return "\n".join([summary + ".", "\nParameters\n---------"] + [
        ".\n".join(params),
        "\nReturns\n-------",
        returns,
    ]) + "."
