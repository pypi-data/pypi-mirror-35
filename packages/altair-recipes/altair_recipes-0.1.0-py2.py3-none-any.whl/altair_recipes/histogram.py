"""Generate histograms."""
import altair as alt
from .common import multivariate_preprocess
from .docstrings import make_docstring


def histogram(data, column, mark={}, encoding={}, properties={}):
    """See below."""
    return alt.Chart(data).mark_bar(**mark).encode(
        alt.X(column + ":Q", bin=True), alt.Y("count()"),
        **encoding).properties(**properties)


histogram.__doc__ = make_docstring(histogram, summary="Generate a histogram")


def layered_histogram(data,
                      columns=None,
                      group_by=None,
                      mark={},
                      encoding={},
                      properties={}):
    """See below."""
    data, key, value = multivariate_preprocess(data, columns, group_by)
    return alt.Chart(data).mark_area(
        opacity=1 / (len(data[group_by].unique())
                     if group_by is not None else len(columns)),
        interpolate="step",
        **mark).encode(
            alt.X(value + ":Q", bin=True), alt.Y("count()", stack=None),
            alt.Color(key + ":N"), **encoding).properties(**properties)


layered_histogram.__doc__ = make_docstring(
    layered_histogram, "Generate multiple overlapping histograms")
