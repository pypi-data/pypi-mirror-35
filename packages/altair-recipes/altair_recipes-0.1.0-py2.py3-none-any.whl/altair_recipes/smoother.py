"""Smoother graph."""
import altair as alt
from .docstrings import make_docstring


def smoother(data,
             x,
             y,
             window=None,
             interquartile_area=True,
             mark={},
             encoding={},
             properties={}):
    """See below."""
    window = data.shape[0] // 4 if window is None else int(window)
    _data = data.sort_values(by=x)
    _data["x"] = _data["x"].rolling(window).median()
    _data["median"] = _data["y"].rolling(window).median()
    chart_line = alt.Chart(_data).mark_line().encode(x=x, y="median")
    if interquartile_area:
        _data["q1"] = _data["y"].rolling(window).quantile(.25)
        _data["q3"] = _data["y"].rolling(window).quantile(.75)
        chart_area = alt.Chart(_data).mark_area(opacity=.2).encode(
            x=x, y="q1", y2="q3")
        return chart_line + chart_area
    else:
        return chart_line


smoother.__doc__ = make_docstring(
    smoother,
    summary="Generate a smooth line plot with optional IRQ shading area",
    additional_params=dict(
        window="""window: int
    The size of the smoothing window""",
        interquartile_area="""interquartile_area: bool
    Whether to plot the IRQ as an area"""))
