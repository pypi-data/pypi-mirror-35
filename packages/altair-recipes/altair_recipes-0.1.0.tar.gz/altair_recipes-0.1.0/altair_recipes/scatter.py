"""Scatter plots."""
import altair as alt
from .common import default
from .docstrings import make_docstring


def scatter(data, x, y, mark={}, encoding={}, properties={}):
    """See below."""

    return alt.Chart(data).mark_point(**mark).encode(
        x=x, y=y, **encoding).properties(**properties)


scatter.__doc__ = make_docstring(scatter, summary="Generate a scatter plot")


def multiscatter(data,
                 columns=None,
                 group_by=None,
                 mark={},
                 encoding={},
                 properties={}):
    """See below."""

    columns = list(default(columns, data.columns))
    assert group_by is None, "Long format not supported yet"
    return alt.Chart(data).mark_point(**mark).encode(
        alt.X(alt.repeat("column"), type='quantitative'),
        alt.Y(alt.repeat("row"), type='quantitative'),
        **encoding).properties(**properties).repeat(
            row=columns, column=columns)


multiscatter.__doc__ = make_docstring(
    multiscatter,
    summary="Generate many scatter plots, all vs. all,\
     from several columns(default all)")
