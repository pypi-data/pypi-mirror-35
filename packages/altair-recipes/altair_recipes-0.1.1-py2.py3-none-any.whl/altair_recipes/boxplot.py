"""Boxplot implementation."""
import altair as alt
from .common import multivariate_preprocess
from .docstrings import make_docstring


#TODO: inject addional option into graph
#TODO: unclear how to do it with complex graphics such as this, if there is a way
def boxplot(data,
            columns=None,
            group_by=None,
            mark={},
            encoding={},
            properties={}):
    """See below."""
    data, key, value = multivariate_preprocess(data, columns, group_by)
    #long form assumed from here
    chart = alt.Chart(data)
    chart_bar = chart.mark_bar(filled=False)
    chart_tick = chart.mark_tick()
    min_value = "min(" + value + ")"
    max_value = "max(" + value + ")"
    median_value = "median(" + value + ")"
    q1_bar = chart_bar.encode(x=key, y="q1(" + value + ")", y2=median_value)
    q3_bar = chart_bar.encode(x=key, y=median_value, y2="q3(" + value + ")")

    min_tick = chart_tick.encode(x=key, y=min_value)
    max_tick = chart_tick.encode(x=key, y=max_value)
    rule = chart.mark_rule().encode(
        x=key, y=alt.Y(min_value, axis=alt.Axis(title=value)), y2=max_value)
    return q1_bar + q3_bar + min_tick + max_tick + rule


boxplot.__doc__ = make_docstring(boxplot, summary="Generate a boxplot")
