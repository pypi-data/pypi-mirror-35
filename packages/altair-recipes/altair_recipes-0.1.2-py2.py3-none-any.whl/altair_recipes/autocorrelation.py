"""Autocorrelation plot."""
import altair as alt
import numpy as np
import pandas as pd
from .common import to_dataframe
from .docstrings import make_docstring


def autocorrelation(data,
                    column,
                    max_lag=None,
                    mark={},
                    encoding={},
                    properties={}):
    """See below."""
    data = to_dataframe(data)
    max_lag = data.shape[0] - 1 if max_lag is None else int(max_lag)
    lags = np.arange(0, max_lag + 1)
    _data = pd.DataFrame(
        dict(
            Lag=lags,
            Autocorrelation=[data[column].autocorr(lag=lag) for lag in lags]))
    return alt.Chart(_data).mark_bar(**mark).encode(
        x="Lag:O", y="Autocorrelation" + ":Q",
        **encoding).properties(**properties)


autocorrelation.__doc__ = make_docstring(
    autocorrelation,
    summary="Generate an autocorrelation plot",
    additional_params=dict(max_lag="""max_lag: int
    Maximum lag to show in the plot, default to number of rows in data"""))
