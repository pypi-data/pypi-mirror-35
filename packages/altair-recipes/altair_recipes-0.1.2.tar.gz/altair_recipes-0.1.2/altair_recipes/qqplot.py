"""Generate qqplots."""
import altair as alt
import numpy as np
from .common import to_dataframe
from .docstrings import make_docstring


def qqplot(data, x, y, mark={}, encoding={}, properties={}):
    """See below."""
    data = to_dataframe(data)
    n = data.shape[0]
    data = data.quantile(np.arange(0, 1, step=10 / n))
    return alt.Chart(data).mark_point(**mark).encode(
        x=x, y=y, **encoding).properties(**properties)


qqplot.__doc__ = make_docstring(
    qqplot, summary="Generate a quantile-quantile plot")
