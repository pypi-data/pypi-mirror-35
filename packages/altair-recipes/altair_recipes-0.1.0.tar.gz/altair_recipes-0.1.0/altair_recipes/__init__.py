# -*- coding: utf-8 -*-
"""Top-level package for altair_recipes."""
from .autocorrelation import autocorrelation
from .boxplot import boxplot
from .histogram import histogram, layered_histogram
from .qqplot import qqplot
from .scatter import scatter, multiscatter
from .smoother import smoother

# alphabetically ordered exports
__all__ = [
    'autocorrelation', 'boxplot', 'histogram', 'layered_histogram',
    'multiscatter', 'qqplot', 'scatter', 'smoother'
]
__author__ = """Antonio Piccolboni"""
__email__ = 'antonio@piccolboni.info'
__version__ = '0.1.0'
