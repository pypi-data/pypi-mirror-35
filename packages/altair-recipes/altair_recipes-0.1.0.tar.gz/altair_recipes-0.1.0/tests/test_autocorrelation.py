import altair_recipes as ar
from altair_recipes.common import viz_reg_test
import numpy as np
import pandas as pd


@viz_reg_test
def test_autocorrelation():
    np.random.seed(0)
    data = pd.DataFrame(dict(x=np.random.uniform(size=100)))
    return ar.autocorrelation(data, "x", max_lag=15)
