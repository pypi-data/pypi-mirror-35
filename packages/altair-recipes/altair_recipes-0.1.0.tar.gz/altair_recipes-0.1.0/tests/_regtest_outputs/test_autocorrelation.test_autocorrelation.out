{
  "$schema": "https://vega.github.io/schema/vega-lite/v2.4.3.json",
  "config": {
    "view": {
      "height": 300,
      "width": 400
    }
  },
  "data": {
    "values": [
      {
        "Autocorrelation": 1.0,
        "Lag": 0
      },
      {
        "Autocorrelation": -0.07051583228220215,
        "Lag": 1
      },
      {
        "Autocorrelation": 0.2825921981910393,
        "Lag": 2
      },
      {
        "Autocorrelation": -0.12163420132397065,
        "Lag": 3
      },
      {
        "Autocorrelation": 0.0745877924560898,
        "Lag": 4
      },
      {
        "Autocorrelation": -0.07349594858870877,
        "Lag": 5
      },
      {
        "Autocorrelation": 0.07101066702233691,
        "Lag": 6
      },
      {
        "Autocorrelation": -0.05056864883617815,
        "Lag": 7
      },
      {
        "Autocorrelation": 0.05543712871851885,
        "Lag": 8
      },
      {
        "Autocorrelation": -0.17478516829165622,
        "Lag": 9
      },
      {
        "Autocorrelation": 0.23721796363578532,
        "Lag": 10
      },
      {
        "Autocorrelation": -0.12159149087355296,
        "Lag": 11
      },
      {
        "Autocorrelation": 0.12712984228112265,
        "Lag": 12
      },
      {
        "Autocorrelation": -0.04220100445795956,
        "Lag": 13
      },
      {
        "Autocorrelation": 0.11602305290330872,
        "Lag": 14
      },
      {
        "Autocorrelation": 0.041769627025596,
        "Lag": 15
      }
    ]
  },
  "encoding": {
    "x": {
      "field": "Lag",
      "type": "ordinal"
    },
    "y": {
      "field": "Autocorrelation",
      "type": "quantitative"
    }
  },
  "mark": "bar"
}